tm2/pkg/sdk/params/keeper.go

```diff
@@ -87,101 +105,254 @@ func (pk ParamsKeeper) Logger(ctx sdk.Context) *slog.Logger {
 
 func (pk ParamsKeeper) Has(ctx sdk.Context, key string) bool {
 	stor := ctx.Store(pk.key)
-	return stor.Has([]byte(key))
-}
-
-func (pk ParamsKeeper) GetRaw(ctx sdk.Context, key string) []byte {
-	stor := ctx.Store(pk.key)
-	return stor.Get([]byte(key))
+	return stor.Has(storeKey(key))
 }
 
 func (pk ParamsKeeper) GetString(ctx sdk.Context, key string, ptr *string) {
-	checkSuffix(key, ".string")
 	pk.getIfExists(ctx, key, ptr)
 }
 
 func (pk ParamsKeeper) GetBool(ctx sdk.Context, key string, ptr *bool) {
-	checkSuffix(key, ".bool")
 	pk.getIfExists(ctx, key, ptr)
 }
 
 func (pk ParamsKeeper) GetInt64(ctx sdk.Context, key string, ptr *int64) {
-	checkSuffix(key, ".int64")
 	pk.getIfExists(ctx, key, ptr)
 }
 
 func (pk ParamsKeeper) GetUint64(ctx sdk.Context, key string, ptr *uint64) {
-	checkSuffix(key, ".uint64")
 	pk.getIfExists(ctx, key, ptr)
 }
 
 func (pk ParamsKeeper) GetBytes(ctx sdk.Context, key string, ptr *[]byte) {
-	checkSuffix(key, ".bytes")
+	pk.getIfExists(ctx, key, ptr)
+}
+
+func (pk ParamsKeeper) GetStrings(ctx sdk.Context, key string, ptr *[]string) {
 	pk.getIfExists(ctx, key, ptr)
 }
 
 func (pk ParamsKeeper) SetString(ctx sdk.Context, key, value string) {
-	checkSuffix(key, ".string")
 	pk.set(ctx, key, value)
 }
 
 func (pk ParamsKeeper) SetBool(ctx sdk.Context, key string, value bool) {
-	checkSuffix(key, ".bool")
 	pk.set(ctx, key, value)
 }
 
 func (pk ParamsKeeper) SetInt64(ctx sdk.Context, key string, value int64) {
-	checkSuffix(key, ".int64")
 	pk.set(ctx, key, value)
 }
 
 func (pk ParamsKeeper) SetUint64(ctx sdk.Context, key string, value uint64) {
-	checkSuffix(key, ".uint64")
 	pk.set(ctx, key, value)
 }
 
 func (pk ParamsKeeper) SetBytes(ctx sdk.Context, key string, value []byte) {
-	checkSuffix(key, ".bytes")
 	pk.set(ctx, key, value)
 }
 
-func (pk ParamsKeeper) getIfExists(ctx sdk.Context, key string, ptr interface{}) {
+func (pk ParamsKeeper) SetStrings(ctx sdk.Context, key string, value []string) {
+	pk.set(ctx, key, value)
+}
+
+func (pk ParamsKeeper) GetRaw(ctx sdk.Context, key string) []byte {
 	stor := ctx.Store(pk.key)
-	bz := stor.Get([]byte(key))
-	if bz == nil {
-		return
+	return stor.Get(storeKey(key))
+}
+
+func (pk ParamsKeeper) SetRaw(ctx sdk.Context, key string, value []byte) {
+	stor := ctx.Store(pk.key)
+	stor.Set(storeKey(key), value)
+}
+
+func (pk ParamsKeeper) GetStruct(ctx sdk.Context, key string, strctPtr interface{}) {
+	parts := strings.Split(key, ":")
+	if len(parts) != 2 {
+		panic("struct key expected format <module>:<struct name>")
 	}
-	err := amino.UnmarshalJSON(bz, ptr)
-	if err != nil {
-		panic(err)
+	moduleName := parts[0]
+	structName := parts[1]
+	if !pk.IsRegistered(moduleName) {
+		panic("unregistered module name")
 	}
+	if structName != "_" {
```

Comment:
TODO add to GetStruct() documentation? 
this is a sanity check but might be good to keep around for now.